src/wasm/wasm-debug.cpp

```diff
@@ -835,31 +835,27 @@ static void updateRanges(llvm::DWARFYAML::Data& yaml,
     auto& range = yaml.Ranges[i];
     BinaryLocation oldStart = range.Start, oldEnd = range.End, newStart = 0,
                    newEnd = 0;
-    // If this was not an end marker, try to find what it should be updated to.
-    if (oldStart != 0 && oldEnd != 0) {
+    // If this is an end marker (0, 0), or an invalid range (0, x) or (x, 0)
+    // then just emit it as it is - either to mark the end, or to mark an
+    // invalid entry.
```

Comment:
we might also want to consider (x, x) or (x, y) where y < x (i.e. empty ranges). I think those would be similar cases to invalid/DCE'd ranges.

Reply:
Sounds good to me too. Do we know which would be best supported by debuggers?

(I'll land this meanwhile as it fixes breakage; we can investigate other options separately.)